test/suite/test_las07.py

```diff
@@ -0,0 +1,142 @@
+        ds.populate()
+
+        uri2 = "table:las02_extra"
+        ds2 = SimpleDataSet(self, uri2, 0, key_format="S", value_format="S")
+        ds2.populate()
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        bigvalue = "aaaaa" * 100
+        self.large_updates(uri, bigvalue, ds, nrows, 1)
+
+        # Check that all updates are seen
+        self.check(bigvalue, uri, nrows, 1)
+
+        # Check to see lookaside working with old timestamp
+        bigvalue2 = "ddddd" * 100
+        self.large_updates(uri, bigvalue2, ds, nrows, 100)
+
+        # Force out most of the pages by updating a different tree
+        self.large_updates(uri2, bigvalue, ds2, nrows, 100)
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue, uri, nrows, 1)
+        self.check(bigvalue2, uri, nrows, 100)
+
+        # Pin oldest and stable to timestamp 100.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(100) +
+            ',stable_timestamp=' + timestamp_str(100))
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue2, uri, nrows, 100)
+
+        # Second set of update operations with increased timestamp
+        self.large_updates(uri, bigvalue, ds, nrows, 250)
+        self.large_updates(uri, bigvalue2, ds, nrows, 300)
+
+        # Force out most of the pages by updating a different tree
+        self.large_updates(uri2, bigvalue, ds2, nrows, 300)
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue, uri, nrows, 250)
+        self.check(bigvalue2, uri, nrows, 300)
+
+        # Pin oldest and stable to timestamp 300.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(300) +
+            ',stable_timestamp=' + timestamp_str(300))
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue2, uri, nrows, 300)
+
+        # Third set of update operations with increased timestamp
+        self.large_updates(uri, bigvalue, ds, nrows, 350)
+        self.large_updates(uri, bigvalue2, ds, nrows, 400)
+
+        # Force out most of the pages by updating a different tree
+        self.large_updates(uri2, bigvalue, ds2, nrows, 400)
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue, uri, nrows, 350)
+        self.check(bigvalue2, uri, nrows, 400)
+
+        # Pin oldest and stable to timestamp 400.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(400) +
+            ',stable_timestamp=' + timestamp_str(400))
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue2, uri, nrows, 400)
+
+if __name__ == '__main__':
+    wttest.run()
```

Comment:
Is there any reason we couldn't read the LAS table from the test and check values explicitly? (I honestly don't know, I'm just wondering if that would be a useful thing to do.)

And are we testing chains of modify records anywhere? That seems like the most complicated case sweep has to deal with.

Reply:
The sweep server cannot be triggered from the test to let it clean the obsolete records. And also reading from the LAS for correctness is difficult as we are not sure what records it should have cleaned in the previous sweep cycle. There are no stats available currently to find out how many LAS records are available. I feel those are needed.

And regarding `modify` records, I modified the test to include them.